src/datamodel/serializer/Uuid.cpp

```diff
@@ -10,3 +10,122 @@
+	uint8_t ver = (uuid.msb() >> 12) & 0xF;                     // 4 bit
+	uint16_t counter = uuid.msb() & 0xFFF;                      // 12 bit
+	uint8_t var = (uuid.lsb() >> 62) & 0x3;                     // 2 bit
+	uint64_t rand_b = uuid.lsb() & 0x3FFFFFFFFFFFFFFF;          // 62 bit
+
+	// Formatting the UUIDv8 in the traditional format
+	std::stringstream ss;
+	ss << std::hex << std::setfill('0') << std::setw(8)
+	   << ((unix_ts_ms >> 16) & 0xFFFFFFFF)  // First 32 bits of timestamp
+	   << "-" << std::setw(4)
+	   << ((unix_ts_ms)&0xFFFF)  // Next 16 bits of timestamp i.e. last 16 bit
+	                             // of ts
+	   << "-" << std::setw(4)
+	   << (((ver & 0xF) << 12) |
+	       (counter & 0xFFF))  // Last 16 bits of timestamp and version
+	   << "-" << std::setw(4)
+	   << (((var & 0x3) << 14) |
+	       ((rand_b >> 48) & 0x3FFF))  // Variant and randb
+	   << "-" << std::setw(12) << (rand_b & 0xFFFFFFFFFFFF);  // Random number
+
+	return std::move(ss).str();
+}
+
+uprotocol::v1::UUID AsString::deserialize(const std::string& str) {
+	// Check if the UUID string is in the correct format
+	if (str.length() != 36 || str[8] != '-' || str[13] != '-' ||
+	    str[18] != '-' || str[23] != '-') {
+		throw std::invalid_argument("Invalid UUID string format");
+	}
+
+	uprotocol::v1::UUID uuid;
+	uint64_t unix_ts_ms = 0;
+	uint8_t ver = 0;
+	uint16_t counter = 0;
+	uint8_t var = 0;
+	uint64_t rand_b = 0;
+
+	try {
+		// Extract the parts from the UUID string
+		unix_ts_ms = std::stoull(str.substr(0, 8), nullptr, 16) << 16;
+		unix_ts_ms |= std::stoull(str.substr(9, 4), nullptr, 16);
+
+		uint16_t ver_counter = std::stoul(str.substr(14, 4), nullptr, 16);
+		ver = (ver_counter >> 12) & 0xF;
+		counter = ver_counter & 0xFFF;
+
+		uint16_t var_randb = std::stoul(str.substr(19, 4), nullptr, 16);
+		var = (var_randb >> 14) & 0x3;
+		rand_b = static_cast<uint64_t>(var_randb & 0x3FFF) << 48;
+		rand_b |= std::stoull(str.substr(24), nullptr, 16);
+	} catch (const std::exception& e) {
+		throw std::invalid_argument("Invalid UUID string format");
+	}
+
+	// Reconstruct the UUID
+	uuid.set_msb((unix_ts_ms << 16) | (ver << 12) | counter);
+	uuid.set_lsb((static_cast<uint64_t>(var) << 62) | rand_b);
+
+	return uuid;
+}
+
+// Serialization function
+std::vector<uint8_t> AsBytes::serialize(const v1::UUID uuid) {
+	std::vector<uint8_t> bytes(16);
+
+	uint32_t msb_high = htonl(static_cast<uint32_t>(uuid.msb() >> 32));
+	uint32_t msb_low = htonl(static_cast<uint32_t>(uuid.msb() & 0xFFFFFFFF));
+	uint32_t lsb_high = htonl(static_cast<uint32_t>(uuid.lsb() >> 32));
+	uint32_t lsb_low = htonl(static_cast<uint32_t>(uuid.lsb() & 0xFFFFFFFF));
+
+	std::memcpy(bytes.data(), &msb_high, sizeof(uint32_t));
+	std::memcpy(bytes.data() + sizeof(uint32_t), &msb_low, sizeof(uint32_t));
+	std::memcpy(bytes.data() + 2 * sizeof(uint32_t), &lsb_high,
+	            sizeof(uint32_t));
+	std::memcpy(bytes.data() + 3 * sizeof(uint32_t), &lsb_low,
+	            sizeof(uint32_t));
```

Comment:
I know what you're doing, but this is a bit unclear.  Is there no bytes.data[4] way of indexing this array ?
I know these are 32-bit values being manipulated, so fixed offsets 0, 4, 8, 12 wouldn't be confusing to anyone.
I might even go so far as 
constexpr int MSB_HIGH=0
constexpr int MSB_LOW_ =4    (extra underscore to make the constants the same length so the code lines up better?)

I'm less firm about this change, so use your best judgement.